lib/presentation/widget/biometric_authentication/biometric_authentication_setting_viewmodel.dart

```diff
@@ -0,0 +1,143 @@
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'package:domain/domain.dart';
+import 'package:flutter/material.dart';
+import 'package:linshare_flutter_app/presentation/localizations/app_localizations.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/biometric_authentication_setting_action.dart';
+import 'package:linshare_flutter_app/presentation/redux/states/app_state.dart';
+import 'package:linshare_flutter_app/presentation/util/router/app_navigation.dart';
+import 'package:linshare_flutter_app/presentation/widget/base/base_viewmodel.dart';
+import 'package:redux/redux.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/list_biometric_kind_extension.dart';
+
+class BiometricAuthenticationSettingViewModel extends BaseViewModel {
+  final AppNavigation _appNavigation;
+  final AuthenticationBiometricInteractor _authenticationBiometricInteractor;
+  final EnableBiometricInteractor _enableBiometricInteractor;
+  final GetAvailableBiometricInteractor _getAvailableBiometricInteractor;
+  final GetBiometricSettingInteractor _getBiometricSettingInteractor;
+  final DisableBiometricInteractor _disableBiometricInteractor;
+
+  BiometricAuthenticationSettingViewModel(
+    Store<AppState> store,
+    this._appNavigation,
+    this._authenticationBiometricInteractor,
+    this._enableBiometricInteractor,
+    this._getAvailableBiometricInteractor,
+    this._getBiometricSettingInteractor,
+    this._disableBiometricInteractor
+  ) : super(store);
+
+  void getBiometricSetting() {
+    store.dispatch((Store<AppState> store) async {
```

Comment:
You should remove this line, you are not dispatching an action

Reply:
I do not think so. I just want it to run in redux's thread.

Reply:
Well my bad you just create a thunkaction, could you please share me the logs : how does the redux logger display it